tensorflow/core/kernels/data/parallel_interleave_dataset_op_test.cc

```diff
@@ -35,18 +36,21 @@ class ParallelInterleaveDatasetOpTest : public DatasetOpsTestBase {
 
   // Creates a new `ParallelInterleaveDataset` op kernel
   Status CreateParallelInterleaveDatasetKernel(
-      const FunctionDefHelper::AttrValueWrapper &func,
-      const DataTypeVector &output_types,
-      const std::vector<PartialTensorShape> &output_shapes, bool sloppy,
-      std::unique_ptr<OpKernel> *op_kernel) {
+      const FunctionDefHelper::AttrValueWrapper& func,
+      const DataTypeVector& output_types,
+      const std::vector<PartialTensorShape>& output_shapes, bool sloppy,
+      std::unique_ptr<OpKernel>* op_kernel) {
     NodeDef node_def = test::function::NDef(
-        kNodeName, kOpName,
-        {"input_dataset", "cycle_length", "block_length", "num_parallel_calls"},
-        {{"f", func},
-         {"Targuments", {}},
-         {"output_types", output_types},
-         {"output_shapes", output_shapes},
-         {"sloppy", sloppy}});
+        kNodeName, "ParallelInterleaveDatasetV2",
```

Comment:
this should rely on name_utils::OpName

Reply:
There is a similar issue in the test for [PaddedBatchDatasetOpTest](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/data/padded_batch_dataset_op_test.cc#L101). The reason that `PaddedBatchDatasetOpTest` does not rely on `name_utils::OpName` is that `name_utils::OpName` will return `PaddedBatchDataset` instead of `PaddedBatchDatasetV2` (which is the required op name for creating the op kernel). Do you have any suggestion for the version suffix? I will change  both based on your suggestion.

Reply:
The PaddedBatchDatasetOp kernel is used for both V1 and V2 versions of the op. The `op_version_` member (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/data/padded_batch_dataset_op.cc#L368) of the PaddedBatchDatasetOp class can be used to determine the op version. The op should be using "PaddedBatch" for V1 and "PaddedBatchV2" for V2.

In contrast, the ParallelIInterleaveDatasetOp kernel is only used for the V2 version of the op. The V1 version of the op is implemented using the ParallelInterleaveDatasetOp in the experimental directory.

Makes sense?

Reply:
@jsimsa Thanks for the detailed explanation! Yes, it makes sense.